app/connectors/MessageConnector.scala

```diff
@@ -30,56 +28,65 @@ import models.MessageSender
 import models.MessageType
 import models.MovementMessageWithStatus
 import models.TransitWrapper
-import uk.gov.hmrc.http.HttpReads.Implicits.readRaw
+import play.api.http.ContentTypes
+import play.api.http.HeaderNames
 import uk.gov.hmrc.http.HeaderCarrier
-import uk.gov.hmrc.http.HttpResponse
 import uk.gov.hmrc.http.HttpClient
+import uk.gov.hmrc.http.HttpReads.Implicits._
+import uk.gov.hmrc.http.UpstreamErrorResponse
 import utils.Format
 
+import java.time.OffsetDateTime
 import scala.concurrent.ExecutionContext
 import scala.concurrent.Future
+import config.Constants
 
-class MessageConnector @Inject()(config: AppConfig, http: HttpClient, val metrics: Metrics)(implicit ec: ExecutionContext) extends HasMetrics {
+class MessageConnector @Inject() (config: AppConfig, http: HttpClient, val metrics: Metrics)(implicit ec: ExecutionContext) extends HasMetrics {
 
-  def post(arrivalId: ArrivalId, message: MovementMessageWithStatus, dateTime: OffsetDateTime, channelType: ChannelType)(
-    implicit headerCarrier: HeaderCarrier
+  def post(arrivalId: ArrivalId, message: MovementMessageWithStatus, dateTime: OffsetDateTime, channelType: ChannelType)(implicit
+    hc: HeaderCarrier
   ): Future[EisSubmissionResult] = {
 
-    val xmlMessage = TransitWrapper(message.message).toString
-
     val url = config.eisUrl
 
-    lazy val messageSender = MessageSender(arrivalId, message.messageCorrelationId)
+    val xmlMessage = TransitWrapper(message.message).toString
+
+    val messageSender = MessageSender(arrivalId, message.messageCorrelationId)
 
-    val newHeaders = headerCarrier
-      .copy(authorization = None)
-      .withExtraHeaders(addHeaders(message.messageType, dateTime, messageSender, channelType): _*)
+    val requestHeaders =
+      hc.headers(Seq(Constants.XClientIdHeader, Constants.XRequestIdHeader)) ++ messageHeaders(message.messageType, dateTime, messageSender, channelType)
```

Comment:
So if I read this right, it only needs two headers from the original hc?

Reply:
Yes I think so - the eis-router adds its own Authorization header because there is a specific bearer token required for each EIS instance. The other one that we pass around everywhere is for determining which channel the movement came from (the frontend services or the API), which is not so relevant for the router.